x-pack/plugins/data_enhanced/server/search/session/check_running_sessions.ts

```diff
@@ -37,13 +37,15 @@ function isSessionStale(
   logger: Logger
 ) {
   const curTime = moment();
-  // Delete if a running session wasn't polled for in the last notTouchedTimeout OR if a completed session wasn't saved for within onScreenTicompletedTimeoutmeout
+  // Delete if a running session wasn't polled for in the last notTouchedTimeout OR
+  // if a completed \ errored \ canceled session wasn't saved for within completedTimeout
   return (
     (session.attributes.status === SearchSessionStatus.IN_PROGRESS &&
       curTime.diff(moment(session.attributes.touched), 'ms') >
         config.notTouchedTimeout.asMilliseconds()) ||
-    curTime.diff(moment(session.attributes.created), 'ms') >
-      config.completedTimeout.asMilliseconds()
+    (session.attributes.status !== SearchSessionStatus.IN_PROGRESS &&
+      curTime.diff(moment(session.attributes.touched), 'ms') >
+        config.completedTimeout.asMilliseconds())
```

Comment:
We discussed that `completedTimeout` could be misleading because it also could become `error`, `expired`, or `canceled`. 

maybe `settledTimeout ` ?

Reply:
Changed to `notTouchedTimeout` and `notTouchedInProgressTimeout`